js/main.js

```diff
@@ -6,28 +6,45 @@ var $entriesPage = document.querySelector('#entries-page');
 var $new = document.querySelector('#new');
 var $entriesButton = document.querySelector('#entries-button');
 var $entries = document.querySelector('#entries');
+var $li = $entries.childNodes;
 
 $photoUrl.addEventListener('input', function (event) {
   $placeholder.style.setProperty('background-image', 'url(' + event.target.value + ')');
 });
 
 $entryForm.addEventListener('submit', function (event) {
   event.preventDefault();
-  var inputValues = {};
-  inputValues.title = $formId.title.value;
-  inputValues.photoUrl = $formId.photoUrl.value;
-  inputValues.notes = $formId.notes.value;
-  inputValues.nextEntryId = data.nextEntryId++;
-  $placeholder.style.setProperty('background-image', 'url(/images/placeholder-image-square.jpg)');
-  $formId.reset();
-  data.entries.unshift(inputValues);
-  $entriesPage.classList.remove('hidden');
-  $entryForm.classList.add('hidden');
-  $entries.prepend(renderEntry(inputValues));
+  if (data.editing === null) {
+    var inputValues = {};
+    inputValues.title = $formId.title.value;
+    inputValues.photoUrl = $formId.photoUrl.value;
+    inputValues.notes = $formId.notes.value;
+    inputValues.nextEntryId = data.nextEntryId++;
+    $placeholder.style.setProperty('background-image', 'url(/images/placeholder-image-square.jpg)');
+    $formId.reset();
+    data.entries.unshift(inputValues);
+    $entriesPage.classList.remove('hidden');
+    $entryForm.classList.add('hidden');
+    $entries.prepend(renderEntry(inputValues));
+  } else if (data.editing !== null) {
+    var updatedValue = {};
+    updatedValue.title = $formId.title.value;
+    updatedValue.photoUrl = $formId.photoUrl.value;
+    updatedValue.notes = $formId.notes.value;
+    updatedValue.nextEntryId = data.editing.nextEntryId;
+    data.entries.splice(data.entries.length - data.editing.nextEntryId, 1, updatedValue);
+    $entriesPage.classList.remove('hidden');
+    $entryForm.classList.add('hidden');
+    var entryLocation = $li.item(data.entries.length - data.editing.nextEntryId);
```

Comment:
this is what you want to change. Once you start deleting entries, the numbers won't align in this way. Instead, you want to loop through all of the `li` elements and find the one with the matching entryid so you know which one to replace.